src/simtools/db/db_handler.py

```diff
@@ -180,66 +172,103 @@ def _find_latest_simulation_model_db(self):
         model_version: str
             Version of the model.
         collection: str
-            collection of array element (e.g. telescopes, calibration_devices)
-        only_applicable: bool
-            If True, only applicable parameters will be read.
+            Collection of array element (e.g. telescopes, calibration_devices).
 
         Returns
         -------
         dict containing the parameters
-
         """
-        _site, _array_element_name, _model_version = self._validate_model_input(
-            site, array_element_name, model_version
-        )
-        array_element_list = db_array_elements.get_array_element_list_for_db_query(
-            _array_element_name, self, _model_version, collection
+        production_table = self._read_production_table_from_mongo_db(collection, model_version)
+        array_element_list = self._get_array_element_list(
+            array_element_name, site, production_table, collection
         )
+
         pars = {}
         for array_element in array_element_list:
-            _array_elements_cache_key = self._parameter_cache_key(
-                site, array_element, model_version, collection
+            cache_key, cache_dict = self._read_cache(
+                DatabaseHandler.model_parameters_cached,
+                names.validate_site_name(site) if site else None,
+                array_element,
+                model_version,
+                collection,
             )
+            if cache_dict:
+                self._logger.debug(f"Found {array_element} in cache (key: {cache_key})")
+                pars.update(cache_dict)
+                continue
+            self._logger.debug(f"Did not find {array_element} in cache (key: {cache_key})")
+
             try:
-                pars.update(DatabaseHandler.model_parameters_cached[_array_elements_cache_key])
-            except KeyError:
-                pars.update(
-                    self.read_mongo_db(
-                        self.mongo_db_config.get("db_simulation_model", None),
-                        array_element_name=array_element,
-                        model_version=_model_version,
-                        collection_name=collection,
-                        run_location=None,
-                        write_files=False,
-                        only_applicable=only_applicable,
-                    )
-                )
-            DatabaseHandler.model_parameters_cached[_array_elements_cache_key] = pars
+                parameter_version_table = production_table["parameters"][array_element]
+            except KeyError:  # allow missing array elements (parameter dict is checked later)
+                continue
+            DatabaseHandler.model_parameters_cached[cache_key] = self._read_mongo_db(
+                query=self._get_query_from_parameter_version_table(
+                    parameter_version_table, array_element, site
+                ),
+                collection_name=collection,
+            )
```

Comment:
(Sorry, I missed this file when I did the first pass of the review.)

Do I understand correctly that in this line you read from MongoDB and execute the query once per array element? Would it be more efficient to build a larger query and return all array elements with one call?

Reply:
Yes, we call query the DB per array element - this is what we have always done. I don't know what is more efficient (I could imagine that simpler queries are faster). What I want to implement is to get a defined set of parameters for all array elements with one query (e.g., get all array element positions). This is beyond this PR - but we need to test DB query performance again and if necessary improve it.

Reply:
Discussed in person, all good.